inst/@sym/function_handle.m

```diff
@@ -115,7 +115,10 @@
       param.fname = param.fname(1:end-2);
     end
 
-    fname2 = param.fname; fcnname = param.fname;
+    [fcnpath, fcnname, fcnext] = fileparts(param.fname);
+    fcnname (fcnname == '-') = '_';
```

Comment:
let's not do this here: if user wants `-` let her do so...  (We'll fix the tempname thing elsewhere...)